### PR TITLE
docs(feature/pace-51): udf tutorial

### PR DESCRIPTION
**Pull request by bvdeenen**

### CODE REVIEW

File: docs/reference/processing-platform-integrations/databricks.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ PACE leverages Databricks' **Unity Catalog** to create dynamic views based on Da
 4. Next, grant the **User** permission to the principal on your desired Databricks Workspace, also through the [Account Console](https://accounts.cloud.databricks.com/workspaces).
 5. In your workspace, either create a new SQL Warehouse, or choose an existing one, and grant usage permission on it to the service principal. Its size can be very small, as it is only used to create views or list tables.
 6. For PACE to be able to list source tables and apply Data Policies through dynamic views, the service principal requires the `USE CATALOG`, `USE SCHEMA` and `SELECT` privileges on all desired **source** resources. The principal requires `CREATE TABLE` privileges on all **target** schemas where Data Policy views are to be created.
+7. If one wishes to use User Defined Functions, both the PACE service principal, as well as any user on any of the target views where the UDF is being used require an `EXECUTE` permssion. See [the UDF tutorial](/tutorials/udfs.md) for more detail.
```
Comment (ivan-p92) [inline docs/reference/processing-platform-integrations/databricks.md]:
```suggestion
7. If one wishes to use User Defined Functions, the PACE service principal, as well as any user of views where the UDF is used, require an `EXECUTE` permission on the function. See [the UDF tutorial](/tutorials/udfs.md) for more detail.
```

File: docs/tutorials/udfs.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+---
+description: Create and use a Python UDF on Databricks.
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
description: Create and use a Python UDF on Databricks
```

File: docs/tutorials/udfs.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+---
+# User Defined Functions in Python
+In this tutorial we will
+* create a User Defined Function written in Python. We will use DataBricks as a processing platform.
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
* create a User Defined Function written in Python. We will use Databricks as a processing platform.
```

File: docs/tutorials/udfs.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+LANGUAGE PYTHON
+AS $$
+return age*age
+$$;
+```
+NOTE: Make sure both the PACE service credential, as well as any user that might access the resulting SQL VIEW
+on Databricks has `EXECUTE` permissions on the function.
+![exec permissions](./dbr-exec-perms.png)
+
+## Create a data-policy
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
## Create a Data Policy
```

File: docs/tutorials/udfs.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+I've used a demo table with an `age` integer column in it, and downloaded a blueprint data policy
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
We have used a demo table with an `age` integer column in it, and downloaded a blueprint data policy:
```

File: docs/tutorials/udfs.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    pace  get data-policy --processing-platform dbr-pace pace.alpha_test.gddemo > policy.yaml
+
+I've edited the policy file, and included the following field transformation.
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
We have then edited the policy file, and included the following field transformation:
```

File: docs/tutorials/udfs.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  ref: pace.alpha_test.gddemo
+```
+So this field transformation defines that any user (`principals: []`) receives the squared value of the `age` column.
+
+## Create the SQL VIEW on Databricks:
+First `upsert` the policy file to PACE.
+
+    pace upsert data-policy policy.yaml
+
+And the actually apply it on the processing platform.
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
And then actually apply it on the processing platform.
```

File: docs/tutorials/udfs.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        principals: []
+  target:
+    fullname: pace.alpha_test.gddemo_pace_view
+source:
+  ...
+  ref: pace.alpha_test.gddemo
+```
+So this field transformation defines that any user (`principals: []`) receives the squared value of the `age` column.
+
+## Create the SQL VIEW on Databricks:
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
## Create the SQL VIEW on Databricks
```

File: docs/tutorials/udfs.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+First `upsert` the policy file to PACE.
+
+    pace upsert data-policy policy.yaml
+
+And the actually apply it on the processing platform.
+
+    pace apply data-policy pace.alpha_test.gddemo  --processing-platform dbr-pace 
+
+## Investigate the results.
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
## Investigate the results
```

File: docs/tutorials/udfs.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+$$;
+```
+NOTE: Make sure both the PACE service credential, as well as any user that might access the resulting SQL VIEW
+on Databricks has `EXECUTE` permissions on the function.
+![exec permissions](./dbr-exec-perms.png)
+
+## Create a data-policy
+I've used a demo table with an `age` integer column in it, and downloaded a blueprint data policy
+
+    pace  get data-policy --processing-platform dbr-pace pace.alpha_test.gddemo > policy.yaml
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]: Shall we use just `demo` instead of `gddemo`?

File: docs/tutorials/udfs.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+select email, age from pace.alpha_test.gddemo limit 5;
+email                     age
+jeffreypowell@hotmail.com 33
+forbeserik@gmail.com      16
+wboone@gmail.com          64
+oliverjulie@yahoo.com     12
+debra64@hotmail.com       79
+```
+
+But the VIEW not so much :-)
```
Comment (ivan-p92) [inline docs/tutorials/udfs.md]:
```suggestion
But the _view_ clearly shows them squared:
```

**Review by ivan-p92 — CHANGES_REQUESTED** [review]

**Comment by ivan-p92** [comment]
Let's also add a note and reference to UDFs on the field transforms page: https://app.gitbook.com/o/Dr8ojEYmRW7T7TFKEbxI/s/FDBMJIaPCcwljkoFamsY/data-policy/rule-set/field-transform#5.-sql-statement

File: docs/tutorials/udfs.md (hunk 10 above)
Comment (bvdeenen) [reply docs/tutorials/udfs.md]: is goed.

File: docs/tutorials/udfs.md (hunk 11 above)
Comment (bvdeenen) [reply docs/tutorials/udfs.md]: boring :-)

**Pull request merged**